src/store/product/action.js

```diff
@@ -11,17 +12,23 @@ function createProduct(title, price, category, image) {
       type: actionTypes.PRODUCT_REQUEST,
     });
 
-    productService.addProduct({ title, price, category, image }).then(
-      (product) => {
-        dispatch({
-          type: actionTypes.PRODUCT_SUCCESS,
-          product,
-        });
-      },
-      (error) => {
-        alert(error);
-        dispatch({ type: actionTypes.PRODUCT_FAILURE, error });
-      }
-    );
+    productService
+      .addProduct({ title, price, category, image })
+      .then(
+        (product) => {
+          dispatch({
+            type: actionTypes.PRODUCT_SUCCESS,
+            product,
+          });
+        },
+        (error) => {
+          alert(error);
+          dispatch({ type: actionTypes.PRODUCT_FAILURE, error });
+        }
+      )
+      .then(() => {
+        const updatedProducts = JSON.parse(localStorage.getItem('products'));
+        dispatch(productsActions.getProducts(updatedProducts));
```

Comment:
There has to be a better way to do this, maybe by creating my own promise in the product form where I return the new products from local storage and then dispatch the action to create the new products in state? I'm still working on a better solution here.

Reply:
I would love any feedback on a better way to handle this. I looked into this and it looks like an action can dispatch another action, but Im not happy with it.

https://github.com/reduxjs/redux/issues/1738